jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -74,6 +74,33 @@ def _warn_if_not_found(app, file_lists):
     return found
 
 
+def _parse_as_image(path):
+    """
+    Parse and load a 2D image. ``CCDData`` objects require a unit be defined
+    in the fits header - if none is provided, use a fallback and
+    raise an error.
+    """
+    with fits.open(path) as hdulist:
+
+        header = hdulist[0].header.copy()
+        meta = dict(header)
+
+        wcs = WCS(header)
+
+        try:
+            image_ccd = CCDData.read(path, wcs=wcs)
+        except ValueError as e:
+            if str(e) == "a unit for CCDData must be specified.":
+                logging.warning("No 'BUNIT' defined in the header, using 'Jy'.")
+                image_ccd = CCDData.read(path, unit='Jy', wcs=wcs)
```

Comment:
This is not going to work with GWCS but maybe that is future work...

Reply:
My vote is future work - this wasn't related to this task, I just had to add this to be able to work on this issue because it was broken.